docs/source/manual/configuration.rst

```diff
@@ -767,6 +767,126 @@ Name                   Default      Description
 type                   REQUIRED     The filter type.
 ====================== ===========  ================
 
+.. _man-configuration-json-layout:
+
+JSON layout
+-----------
+
+.. code-block:: yaml
+
+    layout:
+      type: json
+      timestampFormat: "yyyy-MM-dd'T'HH:mm:ss.SSSZ"
+      prettyPrint: false
+      appendLineSeparator: true
+      includes:
+        - timestamp
+        - threadName
+        - level
+        - loggerName
+        - message
+        - mdc
+        - exception
+      customFieldNames:
+        timestamp: "@timestamp"
+      additionalFields:
+        service-name: "user-service"
+
+=======================  =====================  ================
+Name                     Default                Description
+=======================  =====================  ================
+timestampFormat          (none)                 By default, the timestamp is not formatted. To customize how timestamps are formatted,
+                                                set the property to the corresponding ``DateFormatter`` string or one of the
+                                                ``DateFormatter``'s predefined formats (e.g. ``ISO_LOCAL_TIME``, ``ISO_ZONED_DATE_TIME``,
+                                                ``RFC_1123_DATE_TIME``).
+prettyPrint              false                  Whether the JSON output should be formatted for human readability.
+appendLineSeparator      true                   Whether to append a line separator at the end of the message formatted as JSON.
+includes                 (timestamp, level,
+                          threadName,  mdc,
```

Comment:
Do you want to support mdc filtering? I can see both sides, include all mdc keys or support a whitelist of mdc keys. Generally I think you would want to include all in the case of dynamic diagnostics, but there may users that are abusing MDC as a way to share state. This is something that can be addressed in a follow up if there is demand, as long as that change could be passively introduced later.

Reply:
Looks like a good idea. https://github.com/logstash/logstash-logback-encode and https://github.com/mp911de/logstash-gelf support filtering of MDC, so it makes sense to us to support it too.